File: packages/artifacts-py/src/tigr_contract_artifacts/access.py
from __future__ import annotations
from importlib.resources import files
from pathlib import Path
import json
import yaml

def _packaged_root() -> Path:
    return files("tigr_contract_artifacts") / "_contract"

def _repo_root() -> Path:
    return Path(__file__).resolve().parents[4] / "contract"

def contract_root() -> Path:
    packaged = Path(_packaged_root())
    if packaged.exists():
        return packaged
    return _repo_root()

def load_yaml(relpath: str):
    return yaml.safe_load((contract_root() / relpath).read_text(encoding="utf-8"))

def load_json(relpath: str):
    return json.loads((contract_root() / relpath).read_text(encoding="utf-8"))

def manifest():
    return load_json("manifest.json")

def checksums() -> str:
    return (contract_root() / "checksums.txt").read_text(encoding="utf-8")

File: packages/artifacts-rs/src/access.rs
use std::fs;
use std::path::{Path, PathBuf};

pub fn contract_root() -> PathBuf {
    let crate_root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
    let vendored = crate_root.join("contract");
    if vendored.exists() {
        return vendored;
    }
    crate_root.join("../../contract")
}

pub fn load_json(rel: &str) -> serde_json::Value {
    let path = contract_root().join(rel);
    let data = fs::read_to_string(path).expect("read json");
    serde_json::from_str(&data).expect("parse json")
}

pub fn load_yaml(rel: &str) -> serde_yaml::Value {
    let path = contract_root().join(rel);
    let data = fs::read_to_string(path).expect("read yaml");
    serde_yaml::from_str(&data).expect("parse yaml")
}

pub fn manifest() -> serde_json::Value {
    load_json("manifest.json")
}

pub fn checksums() -> String {
    fs::read_to_string(contract_root().join("checksums.txt")).expect("read checksums")
}
